ext/strings_test.go

```diff
@@ -99,6 +99,14 @@ var stringTests = []struct {
 	{expr: `['x', 'y'].join('-') == 'x-y'`},
 	{expr: `[].join() == ''`},
 	{expr: `[].join('-') == ''`},
+	// Escaping tests.
+	{expr: `"first\nsecond".escape() == "\"first\\nsecond\""`},
+	{expr: `"bell\a".escape() == "\"bell\\a\""`},
+	{expr: `"\bbackspace".escape() == "\"\\bbackspace\""`},
+	{expr: `"\fform feed".escape() == "\"\\fform feed\""`},
+	{expr: `"carriage \r return".escape() == "\"carriage \\r return\""`},
+	{expr: `"horizontal tab\t".escape() == "\"horizontal tab\\t\""`},
+	{expr: `"vertical \v tab".escape() == "\"vertical \\v tab\""`},
```

Comment:
Missing test cases for:
1. string has `\\`
2. string has more than two `\` together like `\a\n`
3. string has `\x` which does not exist
4. string ends with `\`
5. string does not have `\`
You might be able to come up with more :)

Reply:
Right, I've added all but the 3rd one (non-existent escape sequence), since those are caught at parse-time - for instance, modifying `{expr: `strings.quote("printable unicode😀") == "\"printable unicode😀\""`},` to include a `\j` in the middle gives:

```
--- FAIL: TestStrings (0.04s)
    --- FAIL: TestStrings/[72] (0.00s)
        strings_test.go:302: env.Parse(strings.quote("printable \j unicode😀") == "\"printable unicode😀\"") failed: ERROR: <input>:1:15: Syntax error: token recognition error at: '"printable \j'
             | strings.quote("printable \j unicode😀") == "\"printable unicode😀\""
             | ..............^
            ERROR: <input>:1:36: Syntax error: token recognition error at: '😀'
             | strings.quote("printable \j unicode😀") == "\"printable unicode😀\""
             | ...................................＾
            ERROR: <input>:1:37: Syntax error: mismatched input '") == "' expecting ')'
             | strings.quote("printable \j unicode😀") == "\"printable unicode😀\""
             | ...................................．^
            ERROR: <input>:1:44: Syntax error: token recognition error at: '\'
             | strings.quote("printable \j unicode😀") == "\"printable unicode😀\""
             | ...................................．.......^
FAIL
exit status 1
FAIL    github.com/google/cel-go/ext    0.081s
```

Reply:
(marking as resolved)